### ISSUE TITLE
PSR2 standard not checking if closing paren of single-line function declaration is on new line

### ISSUE THREAD

**Issue by rusheb** [squizlabs/PHP_CodeSniffer#2511]
Initial code:
```php
public function hello(string $greeting
) {
    // body
}
```
---

phpcs shows the following error: `Opening brace should be on a new line`.

I guess it is interpreting it as a multi-line argument list? 

---

"Fixed" incorrectly with `phpcbf --standard=PSR2`:
```php
public function hello(string $greeting
)
{
    // body
}
```
---

Desired fix:
```php
public function hello(string $greeting) {
    // body
}
```

I think the above makes more sense and conforms more closely to the [example from PSR2](https://www.php-fig.org/psr/psr-2/#44-method-arguments).

---

Alternative acceptable fix:
```php
public function hello(
    string $greeting
) {
    // body
}
```

**Comment by gsherwood** [comment]
The problem is that this is *not* being detected as a multi-line function declaration, and there is no check in PSR2 to enforce that the closing parenthesis is on the correct line.

This is primarily because other standards assume a declaration is multi-line when the open/close parentheses are on different lines, but PSR2 enforces things differently: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide-meta.md#31---multi-line-arguments-09082013

What needs to happen here is for one of sniffs (probably the top-level PEAR sniff) to not just check the position of the opening brace for single line definitions, but also the position of the closing parenthesis.

**Comment by gsherwood** [comment]
I've pushed a fix for this. The sample code:
```php
<?php
public function hello(string $greeting
) {
    // body
}
```
Will now produce the following error report for PSR2:
```
FILE: /Users/gsherwood/Sites/Projects/PHP_CodeSniffer/temp.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------
 3 | ERROR | [x] There must not be a newline before the closing parenthesis of a single-line function declaration
   |       |     (Squiz.Functions.MultiLineFunctionDeclaration.CloseBracketNewLine)
 3 | ERROR | [x] Opening brace should be on a new line (Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine)
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------
```
And will be fixed like this:
```php
<?php
public function hello(string $greeting)
{
    // body
}
```
Thanks for reporting this.

**Comment by jrfnl** [comment]
I haven't tested this yet, but I suspect this might just fix some more of the existing fixer conflicts. Will test when I have some time.
